src/scripts/app.ts

```diff
@@ -132,57 +137,63 @@ function createToFindItem() {
 
 const checkElements = (event: Event) => {
 	const clickedImg = event.target as HTMLElement;
-	const imgFind = document.querySelector('.image-to-find') as HTMLImageElement;
 
+	const imgFind = document.querySelector('.image-to-find') as HTMLImageElement;
 	const clickedSrcValue = clickedImg.getAttribute('src');
 	const srcToFindValue = imgFind.getAttribute('src');
 	const clickedIdValue = clickedImg.getAttribute('id');
 	const idToFind = imgFind.getAttribute('id');
 
-	if (attemptCounter >= maxAttempts) {
-		showToast('error', 'Brak dalszych prób');
+	if (ATTEMPT_COUNTER >= MAX_ATTEMPTS) {
+		showToast('error', 'No further attempts, Try Again');
 		return;
 	}
 
-	if (clickedImg.classList.contains('js-game-img')) {
-		attemptCounter++;
+	if (clickedImg.classList.contains('game__image')) {
+		ATTEMPT_COUNTER++;
+		clickedImg.classList.remove('hidden');
 		console.log('clicked od the image', clickedImg);
-		checkMatch(clickedSrcValue, srcToFindValue, idToFind, clickedIdValue);
+
+		if (
+			(clickedSrcValue && srcToFindValue && idToFind && clickedIdValue) !== null
+		) {
+			checkMatchAndShowScoreboard(
+				clickedSrcValue,
+				srcToFindValue,
+				idToFind,
+				clickedIdValue
+			);
```

Comment:
To też musimy zmienić na obiekt, łatwo się tutaj pogubić już i przekazać w złej kolejności lub z pominięciem czegoś.